base/src/org/compiere/process/PSCreateFromHRMovement.java

```diff
@@ -45,20 +52,27 @@ protected void prepare() {
 	protected String doIt() throws Exception {
 		//	Instance current Payment Selection
 		MPaySelection paySelection = new MPaySelection(getCtx(), getRecord_ID(), get_TrxName());
-		m_SeqNo = paySelection.getLastLineNo();
+		sequence.set(paySelection.getLastLineNo());
 		//	Loop for keys
-		for(Integer key : getSelectionKeys()) {
+		getSelectionKeys().forEach(key -> {
 			//	get values from result set
-			int HR_Movement_ID = key;
-			String PaymentRule = getSelectionAsString(key, "HRM_PaymentRule");
-			BigDecimal Amount = getSelectionAsBigDecimal(key, "HRM_Amount");
-			m_SeqNo += 10;
-			MPaySelectionLine line = new MPaySelectionLine(paySelection, m_SeqNo, PaymentRule);
+			int movementId = key;
+			String paymentRule = getSelectionAsString(key, "HRM_PaymentRule");
+			BigDecimal sourceAmount = getSelectionAsBigDecimal(key, "HRM_Amount");
+			BigDecimal convertedAmount = getSelectionAsBigDecimal(key, "HRM_ConvertedAmount");
+			MPaySelectionLine line = new MPaySelectionLine(paySelection, sequence.getAndAdd(10), paymentRule);
 			//	Add Order
-			line.setHRMovement(HR_Movement_ID, Amount);
+			X_HR_Movement payrollMovement = new X_HR_Movement(getCtx(), movementId, get_TrxName());
+			X_HR_Process payrollProcess = payrollProcessMap.get(payrollMovement.getHR_Process_ID());
+			if(!Optional.ofNullable(payrollProcess).isPresent()) {
+				payrollProcess = (X_HR_Process) payrollMovement.getHR_Process();
```

Comment:
@yamelsenih use directly if and default value :

Optional.ofNullable(payrollProcess).ifPresent( payrollProcess => {
}).orElseGet(() =>  payrollMovement.getHR_Process())